inc/columns.class.php

```diff
@@ -75,17 +75,17 @@ public static function register_sortable_columns( $columns ) {
 	 * @since   2.6.3
 	 * @change  2.6.3
 	 *
-	 * @param   \WP_Query $query  Current WordPress query.
+	 * @param   \WP_Comment_Query $query  Current WordPress query.
 	 */
 	public static function set_orderby_query( $query ) {
-		$orderby = $query->get( 'orderby' );
+		$spam_reason = isset( $_GET['orderby'] ) ? wp_unslash( sanitize_text_field( $_GET['orderby'] ) ) : '';
```

Comment:
Prüfung auf empty ist sicherer + evl. kürzen & Redundanz vermeiden.

```php
$orderByQueryKey = 'orderby';
$spam_reason = wp_unslash( sanitize_text_field( (!empty($_GET[$orderByQueryKey]) ? $_GET[$orderByQueryKey] : '') ) );
```
More readable 
```php
$orderByQueryKey = 'orderby';
$orderByQueryVal = !empty($_GET[$orderByQueryKey]) ? $_GET[$orderByQueryKey] : '';
$spam_reason = wp_unslash( sanitize_text_field( $orderByQueryVal ) );
```

and just a bit fancy
```php
$orderByQuery = [
	'key' => 'orderby',
	'val' => !empty($_GET[$orderByQueryKey]) ? $_GET[$orderByQueryKey] : ''
];
$spam_reason = wp_unslash( sanitize_text_field( $orderByQuery['val'] ) );
```

Reply:
Why is `empty` safer than `isset`? Can you elaborate on that?

Reply:
Found a useful explanation here: https://www.virendrachandak.com/techtalk/php-isset-vs-empty-vs-is_null/

Reply:
in this case `$_GET['orderby']` can be set but empty and !isset is true.

Reply:
The benefit of `!empty()` instead of `isset()` would be, you would not execute `wp_unslash()` and `sanitize_text_field()`. Other than that, I see no difference in this case.

Reply:
I also don't see how `empty` would be more secure. I usually always use `isset` as recommended in the WPCS wiki: https://github.com/WordPress/WordPress-Coding-Standards/wiki/Fixing-errors-for-input-data#sanitization

But I just realize that I have the two functions in the wrong order (which would still work) :/

Reply:
In this case it will work.
_sanitize_text_fields will handle that. 
```php
function _sanitize_text_fields( $str, $keep_newlines = false ) {
	if ( is_object( $str ) || is_array( $str ) ) {
		return '';
	}
``` 

But if you have a case or stupid code how not prevent the error if ` $_POST['foo']` empty or null this will trigger a `undefined index` notice.

Reply:
The point of `empty()`It doesn't throw a warning if a variable does not exist or is null.

Reply:
The Superglobal `$_GET` will never be null. Therefore `isset` will also not throw an error, if `$_GET['orderby']` is not set.